libbpf-rs/src/map.rs

```diff
@@ -286,6 +287,11 @@ impl Map {
             Err(Error::System(errno::errno()))
         }
     }
+
+    /// Returns an iterator over keys in this map
+    pub fn keys(&self) -> MapKeyIter {
```

Comment:
it's probably worthwhile to mention that if BPF map is modified during iteration than iteration can effectively restart from the start, or it could skip some keys, or could duplicate some keys. Basically, reliably key iteration requires stable BPF map without deletes and updates.